prowler/providers/kubernetes/kubernetes_provider.py

```diff
@@ -146,42 +149,58 @@ def setup_session(self, kubeconfig_file, input_context) -> KubernetesSession:
             Tuple: A tuple containing the API client and the context.
         """
+        logger.info(f"Using kubeconfig file: {kubeconfig_file}")
         try:
-            logger.info(f"Using kubeconfig file: {kubeconfig_file}")
-            try:
-                config.load_kube_config(
-                    config_file=(
-                        os.path.abspath(kubeconfig_file)
-                        if kubeconfig_file != "~/.kube/config"
-                        else os.path.expanduser(kubeconfig_file)
-                    ),
-                    context=input_context,
-                )
-            except ConfigException:
-                # If the kubeconfig file is not found, try to use the in-cluster config
-                logger.info("Using in-cluster config")
-                config.load_incluster_config()
-                context = {
-                    "name": "In-Cluster",
-                    "context": {
-                        "cluster": "in-cluster",  # Placeholder, as the real cluster name is not available
-                        "user": "service-account-name",  # Also a placeholder
-                    },
-                }
+            config.load_kube_config(
+                config_file=(
+                    os.path.abspath(kubeconfig_file)
+                    if kubeconfig_file != "~/.kube/config"
+                    else os.path.expanduser(kubeconfig_file)
+                ),
+                context=input_context,
+            )
+        except ConfigException:
+            # If the kubeconfig file is not found, try to use the in-cluster config
+            logger.info("Using in-cluster config")
+            config.load_incluster_config()
+            context = {
+                "name": "In-Cluster",
+                "context": {
+                    "cluster": "in-cluster",  # Placeholder, as the real cluster name is not available
+                    "user": "service-account-name",  # Also a placeholder
+                },
+            }
+        else:
+            if input_context:
+                contexts = config.list_kube_config_contexts()[0]
+                for context_item in contexts:
+                    if context_item["name"] == input_context:
+                        context = context_item
             else:
-                if input_context:
-                    contexts = config.list_kube_config_contexts()[0]
-                    for context_item in contexts:
-                        if context_item["name"] == input_context:
-                            context = context_item
-                else:
-                    context = config.list_kube_config_contexts()[1]
-            return KubernetesSession(api_client=client.ApiClient(), context=context)
+                context = config.list_kube_config_contexts()[1]
+        return KubernetesSession(api_client=client.ApiClient(), context=context)
+
+    @staticmethod
+    def test_connection(kubeconfig_file, input_context) -> KubernetesSession:
```

Comment:
```suggestion
    def test_connection(kubeconfig_file: str, input_context: str, raise_on_exception: bool = True) -> Connection:
```

Merge from `master` to get the `Connection` object:
```python

@dataclass
class Connection:
    is_connected: bool = False
    error: Exception = None
```